arlas-core/src/main/java/io/arlas/server/model/response/Hit.java

```diff
@@ -57,18 +58,19 @@ public Hit(CollectionReference collectionReference, Map<String, Object> source,
         if (collectionReference.params.idPath != null) {
             md.id = "" + MapExplorer.getObjectFromPath(collectionReference.params.idPath, source);
         }
+        CollectionReferenceManager.setCollectionGeometriesType(source, collectionReference);
         if (collectionReference.params.centroidPath != null) {
             try {
                 Object m = MapExplorer.getObjectFromPath(collectionReference.params.centroidPath, source);
-                md.centroid = m != null ? GeoTypeMapper.getGeoJsonObject(m) : null;
+                md.centroid = m != null ? GeoTypeMapper.getGeoJsonObject(m, collectionReference.params.getCentroidType()) : null;
             } catch (ArlasException e) {
```

Comment:
- e.printStackTrace();
+ LOGGER.error() 
+ a comment explain why this never happens or why we don't have to throw an exception